packages/jmix-front-generator/src/generators/react-typescript/entity-multi-selection-table/answers.ts

```diff
@@ -0,0 +1,77 @@
+import {StudioTemplateProperty, StudioTemplatePropertyType} from "../../../common/studio/studio-model";
+import {ProjectModel} from "../../../common/model/cuba-model";
+import {YeomanGenerator} from "../../../building-blocks/YeomanGenerator";
+import {CommonGenerationOptions} from "../../../common/cli-options";
+import {EntityWithPath} from "../../../building-blocks/stages/template-model/pieces/entity";
+import {
+  askStringIdQuestions,
+  StringIdAnswers,
+  stringIdQuestions
+} from "../../../building-blocks/stages/answers/pieces/stringId";
+import {askQuestions} from "../../../building-blocks/stages/answers/defaultGetAnswersFromPrompt";
+import {isStringIdEntity} from "../common/entity";
+
+export const commonEntityBrowserQuestions: StudioTemplateProperty[] = [
+  {
+    code: 'entity',
+    caption: 'Entity',
+    propertyType: StudioTemplatePropertyType.ENTITY,
+    required: true
+  },
+  {
+    code: 'componentName',
+    caption: 'Component name',
+    propertyType: StudioTemplatePropertyType.POLYMER_COMPONENT_NAME,
+    defaultValue: "List",
+    required: true
+  },
+  {
+    caption: "Menu item",
+    code: "menuItem",
+    propertyType: StudioTemplatePropertyType.MENU_ITEM,
+    required: false
+  },
+  {
+    code: 'query',
+    // Subject to change, in future we might want to get the full query from Studio
+    caption: 'GraphQL query for entity browser',
+    propertyType: StudioTemplatePropertyType.GRAPHQL_QUERY,
+    relatedProperty: "entity",
+    required: true
+  }
+];
+
+export interface EntityBrowserAnswers extends StringIdAnswers {
+  entity: EntityWithPath;
+  componentName: string;
+  query: string;
+  menuItem: string | null;
+}
+
+export const allQuestions: StudioTemplateProperty[] = [
+  ...commonEntityBrowserQuestions,
+];
+
+export const getAnswersFromPrompt = async (
+  projectModel: ProjectModel, gen: YeomanGenerator, _options: CommonGenerationOptions
+): Promise<EntityBrowserAnswers> => {
+  const initialQuestions = [
+    ...commonEntityBrowserQuestions
+  ];
+
+  const answers: EntityBrowserAnswers = await askQuestions<EntityBrowserAnswers>(initialQuestions, projectModel, gen);
+
+  if (isStringIdEntity(projectModel, answers.entity)) {
+    const stringIdAnswers = await askStringIdQuestions(
+      answers.entity,
+      projectModel,
+      gen
+    );
+    return {
+      ...answers,
+      ...stringIdAnswers
+    }
+  }
+
+  return answers;
+};
```

Comment:
Looks like we are duplicating a lot of code here. Can we move it to `building-blocks` and reuse from there?